### DISCUSSION TITLE
PostgresSql connection management

### DISCUSSION THREAD

**Discussion by lazyclaw-labs** [linq2db/linq2db#3452] [General]
I have an api project based on SQL server, recently moved to PostgresSql. As expected, it crashed. :-)
I found out that the datacontext, the dataconnection, was not closed after each Linq operation, i.e. Insert. But it's closed when I use SQL server. In PostgresSql, I have to dispose every datacontext I created. Or else, I will have exausted all the available connection in the pool in a matter of 1 minute.

So what is the official word here? I never dispose the data context before, never, since Linq2SQL time. Any reason PostgresSQL behave differently?

**Comment by MaceWindu** [comment]
You should always dispose data context and data connection. Only idea why it worked before is that you used `DataContext` with `CloseAfterUse` set to `true`
